affinity/cpp-20/d0796r2.md

```diff
@@ -170,150 +187,134 @@ This feature could be easily scaled to heterogeneous and distributed systems, as
-
-```cpp
-auto &resources = execution::this_system::resources();
-
-execution::execution_context execContext(resources[0]);
-
-auto &systelLevelResource = execContext.resource();
-
-// resource[0] should be equal to execResource
-
-for (auto res : systelLevelResource.resources()) {
-    std::cout << res.name()  << `\n`;
-}
-```
-*Listing 4: Example of constructing an execution context from an execution resource*
-
 ### Binding execution and allocation to resources
 
-When creating an `execution_context` from a given `execution_resource`, the executors and allocators associated with it are bound to that `execution_resource`. For example: when creating an `execution_resource` from a CPU socket resource, all executors associated with the given socket will spawn execution agents with affinity to the socket partition of the system *(Listing 5)*.
-
-```cpp
-auto cList = std::execution::this_system::resources();
-// FindASocketResource is a user-defined function that finds a 
-// resource that is a CPU socket in the given resource list
-auto& socket = findASocketResource(cList);
-execution_contexteC{socket} // Associated with the socket
-auto executor = eC.executor(); // By transitivity, associated with the socket too
-auto socketAllocator = eC.allocator(); // Retrieve an allocator to the closest memory node
-std::vector<int, decltype(socketAllocator)> v1(100, socketAllocator);
-std::generate(par.on(executor), std::begin(v1), std::end(v1), std::rand);
-```
-*Listing 5: Example of allocating with affinity to an execution resource*
-
-The construction of an `execution_context` on a component implies affinity (where possible) to the given resource. This guarantees that all executors created from that `execution_context` can access the resources and the internal data structures requires to guarantee the placement of the processor.
-
-Only developers that care about resource placement need to care about obtaining executors and allocations from the correct `execution_context` object. Existing code for vectors and STL (including the Parallel STL interface) remains unaffected.
+An execution context, such as a thread pool, may be bound to an execution resource.
+Binding enables multiple execution contexts to be created and bound to disjoint
+execution resources so that their respective execution agents do not compete for
+their respective execution resources.
 
-If a particular policy or algorithm requires to access placement information, the resources associated with the passed executor can be retrieved via the link to the `execution_context`.
 
 ## Header `<execution>` synopsis
 
     namespace std {
     namespace experimental {
     namespace execution {
 
-    /* Execution resource */
+    /* Execution resource capable of executing std::thread */
```

Comment:
I don't think we should couple the execution resource type with `std::thread` as this could restrict from using other kinds of threads of execution.

Reply:
Such a co-routines?
Given the limited time I thought best to start with current-standard foundation of `std::thread` and just associate `thread_execution_resource` .  We can branch out into other architectures later.

Reply:
I was thinking of GPU threads, though co-routines would be another example of non-`std::thread` threads of execution. As I understand it SG1 are moving away from a tight coupling between `std::thread` and threads of execution, and instead, having `std::thread` just be one of many ways to invoke threads of execution. I feel we should try to follow this direction in our design.